mobile/library/common/internal_engine.cc

```diff
@@ -160,6 +163,12 @@ envoy_status_t InternalEngine::main(std::shared_ptr<Envoy::OptionsImplBase> opti
               server_->serverFactoryContext(),
               server_->serverFactoryContext().messageValidationVisitor());
           connectivity_manager_ = Network::ConnectivityManagerFactory{generic_context}.get();
+          if (!hasIpV6Connectivity()) {
+            connectivity_manager_->dnsCache()->setIpVersionToRemove(
+                {Network::Address::IpVersion::v6});
+          } else {
+            connectivity_manager_->dnsCache()->setIpVersionToRemove(absl::nullopt);
```

Comment:
do we need to set this?  isn't the default just nullopt?

Reply:
Yeah. Removed the `nullopt`.